### PR TITLE
Remove 99% of deprecated ioutil usage (src/cmd/linuxkit)

### PR DESCRIPTION
**Pull request by dgageot**
Signed-off-by: David Gageot <david.gageot@docker.com>

### CODE REVIEW

File: pkg/metadata/provider_gcp.go [hunk 1]
```diff
@@ -90,12 +89,15 @@ func gcpGet(url string) ([]byte, error) {
 
 // SSH keys:
 // TODO also retrieve the instance keys and respect block
-//      project keys see:
-//      https://cloud.google.com/compute/docs/instances/ssh-keys
+//
+//	project keys see:
+//	https://cloud.google.com/compute/docs/instances/ssh-keys
+//
```
Comment (thaJeztah) [inline pkg/metadata/provider_gcp.go:95]: I think Go gets confused by these, and formats them as a `code` block, so instead, remove the indent, and add a colon (`:`) after `TODO`, that way GoDoc also picks them up correctly; see https://pkg.go.dev/github.com/fluhus/godoc-tricks#Bugs;

```suggestion
// TODO: also retrieve the instance keys and respect block
// project keys see:
// https://cloud.google.com/compute/docs/instances/ssh-keys
//
```

File: pkg/metadata/provider_gcp.go [hunk 2]
```diff
@@ -81,7 +80,7 @@ func gcpGet(url string) ([]byte, error) {
 	if resp.StatusCode != 200 {
 		return nil, fmt.Errorf("GCP: Status not ok: %d", resp.StatusCode)
 	}
-	body, err := ioutil.ReadAll(resp.Body)
+	body, err := io.ReadAll(resp.Body)
```
Comment (thaJeztah) [inline pkg/metadata/provider_gcp.go:83]: You missed adding `"io"` as import in this file

File: pkg/metadata/provider_vultr.go [hunk 3]
```diff
@@ -2,7 +2,6 @@ package main
 
 import (
 	"fmt"
-	"io/ioutil"
```
Comment (thaJeztah) [inline pkg/metadata/provider_vultr.go:5]: need to add `"io"` to imports here

File: pkg/metadata/provider_digitalocean.go [hunk 4]
```diff
@@ -2,7 +2,6 @@ package main
 
 import (
 	"fmt"
-	"io/ioutil"
```
Comment (thaJeztah) [inline pkg/metadata/provider_digitalocean.go:5]: Need to add `"io"` to imports in this file

File: pkg/metadata/provider_hetzner.go [hunk 5]
```diff
@@ -3,7 +3,6 @@ package main
 import (
 	"encoding/json"
 	"fmt"
-	"io/ioutil"
```
Comment (thaJeztah) [inline pkg/metadata/provider_hetzner.go:6]: Same here

File: pkg/metadata/provider_metaldata.go [hunk 6]
```diff
@@ -2,7 +2,6 @@ package main
 
 import (
 	"fmt"
-	"io/ioutil"
```
Comment (thaJeztah) [inline pkg/metadata/provider_metaldata.go:5]: Here as well

File: pkg/metadata/provider_openstack.go [hunk 7]
```diff
@@ -2,7 +2,6 @@ package main
 
 import (
 	"fmt"
-	"io/ioutil"
```
Comment (thaJeztah) [inline pkg/metadata/provider_openstack.go:5]: And here

File: pkg/metadata/provider_scaleway.go [hunk 8]
```diff
@@ -5,7 +5,6 @@ import (
 	"bytes"
 	"errors"
 	"fmt"
-	"io/ioutil"
```
Comment (thaJeztah) [inline pkg/metadata/provider_scaleway.go:8]: Ditto

**Review by thaJeztah — CHANGES_REQUESTED** [review]
Found some files where you forgot to add the `"io"` to imports, and left a comment on formatting the comments (mostly a nit as these functions are not exported)

**Comment by dgageot** [comment]
Sorry for the mess...

**Comment by deitch** [comment]
Again, I would prefer if we split out the changes to the CLI `src/cmd/linuxkit` and the ones to packages `pkg/`. Just better risk management and ability to roll back if we have issues.

**Review by deitch — APPROVED** [review]

**Pull request merged**